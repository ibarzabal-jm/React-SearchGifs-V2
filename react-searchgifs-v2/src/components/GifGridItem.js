import React from 'react'

export const GifGridItem = ( {id, title, url} ) => {

    

    return (
        <div className="card animate__animated animate__rollIn">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

