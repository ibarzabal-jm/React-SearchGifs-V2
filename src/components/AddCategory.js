import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {
    
    const [inputValue, setInputValue] = useState(''); 
    
    const handleInputChange = ( e ) =>{
        setInputValue( e.target.value );

    }

    const handleSubmit = ( e ) => {
        
        e.preventDefault();

        if( inputValue.trim().length > 2){

            setCategories( categorias => [inputValue, ...categorias ] );
            setInputValue('');
        }else{
            
            console.log('Categor√≠as no agregada')
        }

    }

    return (
        <>
            <h4>Ingresa aqui abajo lo que quieras buscar</h4>
            <form onSubmit={handleSubmit}>
                <input
                    type = "text"
                    value = {inputValue}
                    onChange={ handleInputChange }
    
                />

            </form>
        </>
    )
}


AddCategory.propType = {
    setCategories: PropTypes.func.isRequired
}

