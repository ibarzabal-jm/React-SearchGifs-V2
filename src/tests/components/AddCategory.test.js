import React from 'react'
import '@testing-library/jest-dom'

import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory'

describe('Pruebas en <AddCAtegory />', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={setCategories} /> );
    
    beforeEach( () => {
        jest.clearAllMocks();
        
        wrapper = shallow( <AddCategory setCategories={setCategories} /> );
    });
    
    test('debe de mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
    })

    test('debe de cambiar la caja de texto ', () => {
        
        const input = wrapper.find('input');
        const value ='Hola mundo'

        input.simulate('change', { target : { value } } );
    })

    test('NO debe de mostrar la informacion con submit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault: ()=>{} });

        expect( setCategories ).not.toHaveBeenCalled();
    })
    
    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        
        const value ='Hola prueba'
        const input = wrapper.find('input');

        // simular el inputChange
        input.simulate('change', { target : { value } } );
        //simular el submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        // set categories debe ser llamado
        expect( setCategories ).toHaveBeenCalled();
        // el valor del input debe estar ''
        expect(wrapper.find('input').prop('value') ).toBe('');
        
    })

    
    
    
    
})
