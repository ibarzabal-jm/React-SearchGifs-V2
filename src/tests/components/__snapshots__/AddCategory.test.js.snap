// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCAtegory /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h4>
    Ingresa aqui abajo lo que quieras buscar
  </h4>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
